framework/src/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -371,7 +371,7 @@ public void insertValue(FlowExpressions.Receiver r, /*@Nullable*/ V value) {
      */
     private boolean isMonotonicUpdate(FieldAccess fieldAcc, V value) {
         boolean isMonotonic = false;
-        if (!sequentialSemantics) { // only compute if necessary
+        if (!sequentialSemantics) { // only compute if necessary // TODO: this is an optimization that breaks the contract of the method. This check should be outside of this method.
```

Comment:
Can you put this comment on a separate line and explain in more detail why this "optimization" breaks a contract.